website_menu_by_user_status/views/website_templates.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <template id="website_display_menu" name="Conditional display menu"
+                  inherit_id="website.layout"
+                  active="True" customize_show="True">
+            <xpath expr="//ul[@id='top_menu']/t/t" position="replace">
+                <t t-if="(submenu.user_logged and request.uid != 3) or (submenu.user_not_logged and request.uid == 3)">
```

Comment:
Don't use a number here (that can change per database) but `%(base.public_user)d`